lib/sparql_to_sw_solr/language_field.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'searchworks_languages'
+
+module SparqlToSwSolr
+  class InstanceSolrDoc
+    module LanguageField
+
+      private
+
+      def language_values
+        result = []
+        solution_values(language_solutions).each do |soln_val|
+          # TODO: ultimately, should lookup lang uri's RDF and retrieve it (and cache it) and then
+          # get value from "<skos:prefLabel xml:lang="en">" property (or <madsrdf:authoritativeLabel xml:lang="en">)
+          code = soln_val[%r{^http://id.loc.gov/vocabulary/languages/(.*)}, 1]
+          next unless code && !code.empty?
+          value = SEARCHWORKS_LANGUAGES[code]
+          result << (value ? value : code)
+        end
+        result
+      end
+
+      def language_solutions
+        query = "#{BF_NS_DECL}
+          SELECT ?lang WHERE {
+            <#{instance_uri}> bf:instanceOf ?work .
+            ?work bf:language ?lang .
+          }".freeze
+        sparql.query(query)
+      end
+
+      # SPARQL results expected to have "lang" binding
+      def solution_values(solutions)
+        values = []
+        solutions.each_solution do |soln|
+          values << soln.lang.to_s if soln.bindings.keys.include?(:lang)
+        end
+        values
+      end
```

Comment:
Hmm, this smells of something.  Can't put my finger on it.  Might be premature abstraction.